card/spec/lib/card/query_spec.rb

```diff
@@ -8,6 +8,13 @@
     Card::Query.run @query.reverse_merge return: :name, sort: :name
   end
 
+  it 'should not alter original statement' do
+    @query = { right_plus: { name: %w( in tag source ) } }
+    query_clone = @query.deep_clone
+    subject # runs query
+    expect(query_clone).to eq(@query)
+  end
```

Comment:
I was able to reproduce the bug with this testing. I removed the other new tests, because they do a lot more processing for the same effect.  (I'd really like to get our test suite time down)